pkg/api/vm/handler.go

```diff
@@ -626,6 +642,119 @@ func (h *vmActionHandler) removeVolume(ctx context.Context, namespace, name stri
 		Error()
 }
 
+// cloneVM creates a VM which uses volume cloning from the source VM.
+func (h *vmActionHandler) cloneVM(name string, namespace string, input CloneInput) error {
+	vm, err := h.vmCache.Get(namespace, name)
+	if err != nil {
+		return fmt.Errorf("cannot get vm %s/%s, err: %w", namespace, name, err)
+	}
+	newVM := &kubevirtv1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        input.TargetVM,
+			Namespace:   namespace,
+			Annotations: map[string]string{},
+			Labels:      vm.Labels,
+		},
+		Spec: *vm.Spec.DeepCopy(),
```

Comment:
should we also reset the VM's mac address?

Reply:
Yes, I update it. Thank you.